### PR TITLE
Allow list ownership to be transfered

### LINKED ISSUE #26: Checks temporarily become unchecked during save
When changes to a list is saved the altered checks are temporarily unchecked before the list is saved.

### LINKED ISSUE #27: Allow list ownership to be transfered
It should be possible to transfer the ownership of a list to another user

### LINKED ISSUE #31: Update to use changed submitted and verified properties on lists are handled
- `submitted` has become `submittedAt` which is `null` if not submitted and the timestamp of submission otherwise.
  - If a mod or admin submits a list not already owned by them they should be warned that they will be made the owner of that list
- `Verification` is created when a list is verified which contains both `verifiedBy` and `verifiedAt`

Lists also includes a property `status` now:

```ts
type ListStatus = 'open' | 'submitted' | 'verified';
```

Depends on insektionen/cleaning-list-be#25

### PR DESCRIPTION
**Pull request by StelFoog**
- Allows list ownership to be transfered to other users
- Fixes bug where checks become unchecked during save
- Adds `useDebounce` hook

Resolves #27
Resolved #26

- Updates list types to use new structure
- Adds list `status` property
- Adds confirmation when submitting a not-owned list

Resolves #31

### CODE REVIEW

File: src/list/list.model.ts [hunk 1]
```diff
@@ -11,16 +11,32 @@ export type List = {
...
+	| 'fields'
+	| 'colors'
+	| 'responsible'
+	| 'phoneNumber'
+	| 'comment'
+	| 'createdBy'
+	| 'ownedBy'
 >;
 
+export type ListStatus = 'open' | 'submitted' | 'verified';
```
Comment (loffa) [inline src/list/list.model.ts:33]: As in backend, I think enum would look better for this type of variable.

File: src/style/general.scss [hunk 2]
```diff
@@ -33,7 +33,8 @@ a:hover {
 html,
 body {
 	margin: 0;
-	min-width: 320px;
+	// min-width: 320px;
```
Comment (loffa) [inline src/style/general.scss:36]: Left over commented code

File: src/style/general.scss [hunk 3]
```diff
@@ -33,7 +33,8 @@ a:hover {
...
+	min-width: 100%;
 	max-width: 100%;
```
Comment (loffa) [inline src/style/general.scss:38]: Why set both min-width and max-width to 100% ?
Should set width statically instead in that case.

File: src/views/ListList.tsx [hunk 4]
```diff
@@ -10,10 +10,14 @@ import Verified from '../icons/Verified';
 import { useTranslate } from '../util/translation';
 
 function listIcon(list: MinimalList) {
-	if (list.verified) return <Verified size="1.5rem" />;
-	if (list.submitted) return <Submitted size="1.5rem" />;
-
-	return <NotSubmitted size="1.5rem" />;
+	switch (list.status) {
+		case 'open':
```
Comment (loffa) [inline src/views/ListList.tsx:14]: In case the enum gets added it can be used here

**Review by loffa — COMMENTED** [review]
Functinallity looks good, some small code tweaks commented on.

File: src/list/list.model.ts (hunk 1 above)
Comment (StelFoog) [reply src/list/list.model.ts:33]: Discussing enums in [BE pull request](https://github.com/insektionen/cleaning-list-be/pull/24#discussion_r1117936588).

Comment (loffa) [reply src/list/list.model.ts:33]: Agreed! Solved.

**Review by loffa — APPROVED** [review]

**Pull request merged**